tests/test_cli.py

```diff
@@ -131,6 +131,114 @@ def test_cli_data_parsed_by_question_type(
     assert answers["question"] == expected
 
 
+@pytest.mark.parametrize(
+    "type_, answer, expected",
+    [
+        ("str", "string", "string"),
+        ("str", "1", "1"),
+        ("str", "1.2", "1.2"),
+        ("str", "true", "true"),
+        ("str", "null", "null"),
+        ("str", '["string", 1, 1.2, true, null]', '["string", 1, 1.2, true, null]'),
+        ("int", "1", 1),
+        ("float", "1.2", 1.2),
+        ("bool", "true", True),
+        ("bool", "false", False),
+        ("yaml", '"string"', "string"),
+        ("yaml", "1", 1),
+        ("yaml", "1.2", 1.2),
+        ("yaml", "true", True),
+        ("yaml", "null", None),
+        ("yaml", '["string", 1, 1.2, true, null]', ["string", 1, 1.2, True, None]),
+        ("json", '"string"', "string"),
+        ("json", "1", 1),
+        ("json", "1.2", 1.2),
+        ("json", "true", True),
+        ("json", "null", None),
+        ("json", '["string", 1, 1.2, true, null]', ["string", 1, 1.2, True, None]),
```

Comment:
I think we don't need 30 tests here.

Just pass a `data.yml` that contains a few types and you solve all with a single test.